server/aws/mc-lambda.tf

```diff
@@ -0,0 +1,106 @@
+##
+#  Mectrics Collection Lambda
+##
+
+resource "aws_lambda_function" "metrics" {
+  function_name = var.service_name
+  description = var.lambda-description
+  s3_bucket = var.lambda_code
+  s3_key = var.lambda-function-code
+  handler = var.lambda-function-handler
+  runtime = var.lambda-function-runtime
+  role = aws_iam_role.role.arn
+  environment {
+    variables = {
+      dataBucket = "${var.s3_raw_metrics_bucket_name}-${data.aws_caller_identity.current.account_id}"
+      dataKey = aws_kms_key.mykey.id
+      fileLoca = "metrics"
+    }
+  }
+
+  vpc_config {
+    security_group_ids = []
+    subnet_ids = []
+  }
+  tags = {
+    Name = var.service_name
+    Resource = "Lambda"
+    Environment = var.environment
+    (var.billing_tag_key) = var.billing_tag_value
+    Project = var.project
+    Deployment = "Terraform"
+  }
+}
+
+resource "aws_iam_role" "role" {
+  name = "${var.service_name}.role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_cloudwatch_log_group" "metrics" {
+  name = "/aws/lambda/${var.service_name}"
+  retention_in_days = 14
+}
+
+resource "aws_iam_policy" "lambda_logging" {
+  name = var.service_name
+  path = "/"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*",
+      "Effect": "Allow"
+    },
+    {
+      "Sid": "VisualEditor0",
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "kms:Encrypt"
+      ],
+      "Resource": [
+
+        "${aws_kms_key.mykey.arn}",
+        "arn:aws:s3:::${var.s3_raw_metrics_bucket_name}-${data.aws_caller_identity.current.account_id}/*"
+      ]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_logs" {
+  role = aws_iam_role.role.name
+  policy_arn = aws_iam_policy.lambda_logging.arn
+}
+
+resource "aws_lambda_permission" "apigw" {
+  statement_id = "AllowAPIGatewayInvoke"
+  action = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.metrics.function_name
+  principal = "apigateway.amazonaws.com"
+  source_arn = "${aws_api_gateway_rest_api.metrics.execution_arn}/*/*"
```

Comment:
Can you please just reply with what this source arn maps to, is it everything in the api gateway?

Reply:
Currently it is the entire API which only has the one endpoint but I can scope it to the one POST method.